docker-compose.yml

```diff
@@ -19,4 +19,13 @@ services:
     image: redis:6.2.6-bullseye
     network_mode: bridge
     ports: 
-      - "6379:6379"
\ No newline at end of file
+      - "6379:6379"
+  db:
```

Comment:
I know Harry set up Redis here before and I think that makes some sense because we tie into standard Redis APIs and mocking those might be a bit of work and lead us to make mistakes.

However, I'd rather we mock storage than depend on Postgres.

We have full control of what the storage API is, because we wrote it. So mocking it without depending on Postgres is better imo. We don't have to depend on an external service to run these tests, they will run faster, etc.

Reply:
I thought about this and opted to go for this route because if we mocked we'd need to think about how the update function should be mocked on conflict, concurrent executions and potential future more complex functions. Plus the probability of making an error in the mock code.
Regarding test time it went from 3.35s to 4.7s so yes it's a slower (note times on my intel macbook pro)